### PR TITLE
Introduce socket reconnection interval

### PR DESCRIPTION
**Pull request by pabera**
As discussed in https://github.com/kellerza/hassio-sma-em/issues/3, I had trouble using this plugin for Home Assistant because data stop flowing in after 4-5 minutes. This PR implements an automatic reconnect after a given time interval, in my case 3 min. I happy to discuss the implementation and if you like to make changes to it.

### CODE REVIEW

**Comment by kellerza** [comment]
Thanks @pabera, maybe we can make this RECONNECT_INTERVAL configurable? (add it to config.json) and then if it is not specified/0 we just make it some really large value? Then is covers your case and other cases that seem to be ok?

**Comment by pabera** [comment]
I have updated the value to be in the config. It is by default set to 24h which is the large interval you have referred to. In my case, I can change it to 3 minutes.

In another iteration, since you introduced `asyncio`, this should also be possible by scheduling a task once (without a reconnection at all) or a recurring task (when the interval is needed).

File: sma-em/sensors.py [hunk 1]
```diff
@@ -205,7 +206,8 @@ def startup():
                 "u1:min",
             ],
             THRESHOLD: 80,
-            DEBUG: 0,
+            RECONNECT_INTERVAL: 0,
```
Comment (kellerza) [inline sma-em/sensors.py:209]: Not important in production, but used this in local testing

```suggestion
            RECONNECT_INTERVAL: 86400,
```

File: sma-em/run.py [hunk 2]
```diff
@@ -34,24 +44,22 @@ def main():
...
     LOOP.set_debug(True)
 
-    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-    sock.bind(("", BROADCAST_PORT))
-    group = socket.inet_aton(BROADCAST_ADDR)
-    mreq = struct.pack("4sL", group, socket.INADDR_ANY)
-    sock.setsockopt(socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP, mreq)
-    # sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-    # sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)
+    RECONNECT_INTERVAL = sensors.OPTIONS["RECONNECT_INTERVAL"]
```
Comment (kellerza) [inline sma-em/run.py:47]: Lets use the const & default

```suggestion
    RECONNECT_INTERVAL = sensors.OPTIONS.get(sensors.RECONNECT_INTERVAL, 86400)
```

**Review by kellerza — COMMENTED** [review]
I have not tested, but looks good.
Just 2 small suggestions

**Comment by kellerza** [comment]
Thanks!

**Pull request merged**